packages/common/src/state/simple.ts

```diff
@@ -0,0 +1,185 @@
+import { Account, bytesToHex } from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak.js'
+
+import { OriginalStorageCache } from './originalStorageCache.js'
+
+import type {
+  AccountFields,
+  EVMStateManagerInterface,
+  Proof,
+  StorageDump,
+  StorageRange,
+} from '../interfaces.js'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+/**
+ * Simple and dependency-free state manager for basic state access use cases
+ * where a trie or verkle backed state manager is too heavy-weight.
+ *
+ * This state manager comes with the basic state access logic for
+ * accounts, storage and code (put* and get* methods) as well as a simple
+ * implementation of checkpointing but lacks methods implementations of
+ * state root related logic as well as some other non-core functions.
+ *
+ * Functionality provided is sufficient to be used for simple EVM use
+ * cases and the state manager is used as default there.
+ *
+ * For a more full fledged and MPT-backed state manager implementation
+ * have a look at the `@ethereumjs/statemanager` package.
+ */
+export class SimpleStateManager implements EVMStateManagerInterface {
+  public accountStack: Map<PrefixedHexString, Account | undefined>[] = []
+  public codeStack: Map<PrefixedHexString, Uint8Array>[] = []
+  public storageStack: Map<string, Uint8Array>[] = []
+
+  originalStorageCache: {
+    get(address: Address, key: Uint8Array): Promise<Uint8Array>
+    clear(): void
+  }
+
+  constructor() {
+    this.add()
+    this.originalStorageCache = new OriginalStorageCache(this.getContractStorage.bind(this))
+  }
+
+  topA() {
+    return this.accountStack[this.accountStack.length - 1]
+  }
+  topC() {
+    return this.codeStack[this.codeStack.length - 1]
+  }
+  topS() {
```

Comment:
Should these methods be exported publicly? They should also get a more descriptive name e.g. `topStorageStack()`

Reply:
I would like to not export publicly to not further enlarge the API surface we expose, then we have yet another three methods to care for. 🙂 

TBH I really do like the short names, since they are so unintrusive and I feel both from reading the methods itself as well as usages like:

```ts
this.codeStack.push(new Map(this.topC()))
```

It becomes pretty obvious what they do.

So I personally would like to keep.

Won't die on this hill though. 😂

Reply:
Should we mark these `private`? If we decide to keep those as now (so those are `public` methods), then a description should be added, otherwise there are public methods `topS` with no description.

Reply:
Ah, sorry, I always forget this "public as default" concept.

Yes, will do.

Reply:
Personally not a fan of non-explicit names. At least to me it's not immediately obvious what they do, and I don't see the benefit of shortening/acronyms in general. Less of a concern since they're private though.

Reply:
Ok, generally totally on your side, so let's not make a because-of-the-context exception, will rename to something more expressive

Reply:
(will go for `protected` for extensibility reasons)